C/plugins/storage/sqlite/connection.cpp

```diff
@@ -0,0 +1,2366 @@
+		}
+
+		if (tb.HasMember("format"))
+		{
+			// SQLite 3 date format is limited.
+			string new_format;
+			if (applyDateFormat(tb["format"].GetString(),
+					    new_format))
+			{
+				sql.append(", ");
+				// Add the formatted column
+				sql.append(new_format);
+
+				if (tb.HasMember("size"))
+				{
+					// Use JulianDay, with microseconds
+					sql.append(tb["size"].GetString());
+					sql.append(" * round(");
+					sql.append("strftime('%J', ");
+					sql.append(tb["timestamp"].GetString());
+					sql.append(") / ");
+					sql.append(tb["size"].GetString());
+					sql.append(", 6)");
+				}
+				else
+				{
+					sql.append(tb["timestamp"].GetString());
+				}
+				sql.append(")");
+			}
+			else
+			{
+				/**
+				 * No date format found: we should return an error.
+				 * Note: currently if input Json payload has no 'result' member
+				 * raiseError() results in no data being sent to the client
+				 * For the time being we just use JulianDay, with microseconds
+				 */
+				sql.append(", datetime(");
+				if (tb.HasMember("size"))
+				{
+					sql.append(tb["size"].GetString());
+					sql.append(" * round(");
+				}
+				// Use JulianDay, with microseconds
+				sql.append("strftime('%J', ");
+				sql.append(tb["timestamp"].GetString());
+				if (tb.HasMember("size"))
+				{
+					sql.append(") / ");
+					sql.append(tb["size"].GetString());
+					sql.append(", 6)");
+				}
+				else
+				{
+					sql.append(")");
+				}
+				sql.append(")");
+			}
+		}
+		else
+		{
+			sql.append(", datetime(");
+			if (tb.HasMember("size"))
+			{
+				sql.append(tb["size"].GetString());
+				sql.append(" * round(");
+			}
+
+			/*
+			 * Default format when no format is specified:
+			 * - we use JulianDay in order to get milliseconds,
+			 * - with timezone '+00:00' added
+			 */
+			sql.append("strftime('%J', ");
+			sql.append(tb["timestamp"].GetString());
+			if (tb.HasMember("size"))
+			{
+				sql.append(") / ");
+				sql.append(tb["size"].GetString());
+				sql.append(", 6)");
+			}
+			else
+			{
+				sql.append(")");
+			}
+
+			sql.append(") || '+00:00'");
+		}
+
+		sql.append(" AS \"");
+		if (tb.HasMember("alias"))
+		{
+			sql.append(tb["alias"].GetString());
+		}
+		else
+		{
+			sql.append("timestamp");
+		}
+		sql.append('"');
+	}
+	return true;
+}
+
+/**
+ * Process the modifers for limit, skip, sort and group
+ * TODO
+ * Not tested yet with SQLite3
+ */
+bool Connection::jsonModifiers(const Value& payload, SQLBuffer& sql)
+{
+	if (payload.HasMember("timebucket") && payload.HasMember("sort"))
+	{
+		raiseError("query modifiers",
+			   "Sort and timebucket modifiers can not be used in the same payload");
+		return false;
+	}
+
+	if (payload.HasMember("group"))
+	{
+		sql.append(" GROUP BY ");
+		if (payload["group"].IsObject())
+		{
+			const Value& grp = payload["group"];
+			if (grp.HasMember("format"))
+			{
+				/**
+				 * SQLite 3 date format is limited.
+				 * Handle all availables formats here.
+				 */
```

Comment:
Again use SQLBuffer rather than have an intermediate string

Reply:
Yes, in next commits